Sources/WordPressKit/Models/RemoteBlogSettings.swift

```diff
@@ -186,6 +186,182 @@ public class RemoteBlogSettings: NSObject {
+        case gmtOffset = "gmt_offset"
+        case timezoneString = "timezone_string"
+        case settings = "settings"
+        case defaultCategory = "default_category"
+        case defaultPostFormat = "default_post_format"
+        case dateFormat = "date_format"
+        case timeFormat = "time_format"
+        case startOfWeek = "start_of_week"
+        case postsPerPage = "posts_per_page"
+        case commentsAllowed = "default_comment_status"
+        case commentsBlocklistKeys = "blacklist_keys"
+        case commentsCloseAutomatically = "close_comments_for_old_posts"
+        case commentsCloseAutomaticallyAfterDays = "close_comments_days_old"
+        case commentsKnownUsersAllowlist = "comment_whitelist"
+        case commentsMaxLinks = "comment_max_links"
+        case commentsModerationKeys = "moderation_keys"
+        case commentsPagingEnabled = "page_comments"
+        case commentsPageSize = "comments_per_page"
+        case commentsRequireModeration = "comment_moderation"
+        case commentsRequireNameAndEmail = "require_name_email"
+        case commentsRequireRegistration = "comment_registration"
+        case commentsSortOrder = "comment_order"
+        case commentsThreadingEnabled = "thread_comments"
+        case commentsThreadingDepth = "thread_comments_depth"
+        case pingbackOutbound = "default_pingback_flag"
+        case pingbackInbound = "default_ping_status"
+        case relatedPostsAllowed = "jetpack_relatedposts_allowed"
+        case relatedPostsEnabled = "jetpack_relatedposts_enabled"
+        case relatedPostsShowHeadline = "jetpack_relatedposts_show_headline"
+        case relatedPostsShowThumbnails = "jetpack_relatedposts_show_thumbnails"
+        case ampSupported = "amp_is_supported"
+        case ampEnabled = "amp_is_enabled"
+
+        case sharingButtonStyle = "sharing_button_style"
+        case sharingLabel = "sharing_label"
+        case sharingTwitterName = "twitter_via"
+        case sharingCommentLikesEnabled = "jetpack_comment_likes_enabled"
+        case sharingDisabledLikes = "disabled_likes"
+        case sharingDisabledReblogs = "disabled_reblogs"
+    }
+
+    /// Parses details from a JSON dictionary, as returned by the WordPress.com REST API.
+    @objc
+    public init(jsonDictionary json: [String: Any]) {
+        let rawSettings = json[CodingKeys.settings.rawValue] as? [String: Any] ?? [:]
+
+        name = json[CodingKeys.name.rawValue] as? String
+        tagline = json[CodingKeys.tagline.rawValue] as? String
+        privacy = rawSettings[CodingKeys.privacy.rawValue] as? NSNumber
+        // The value here can be a String, so we need a custom conversion
+        languageID = rawSettings.number(forKey: CodingKeys.languageID.rawValue)
+        iconMediaID = rawSettings[CodingKeys.iconMediaID.rawValue] as? NSNumber
+        // The value here can be a String, so we need a custom conversion
+        gmtOffset = rawSettings.number(forKey: CodingKeys.gmtOffset.rawValue)
```

Comment:
Maybe I worry too much, but do you think we should use `number(forKey:)` to parse all properties instead of just these keys, because that's what the original Objective-C uses?.

We happen to know these weird cases, simply because we have one test json file for it. There is no indication in the original Objective-C code that these keys need special treatment. And, may be there are other keys needs a special "string" to "number" parsing, which is not detected by that one single test json file?

Reply:
Also, [the `string(forKey:)` parsing](https://github.com/wordpress-mobile/NSObject-SafeExpectations/blob/3cc5d8cca23f36efe711feadc37e3b0e13318d8c/Sources/NSObject-SafeExpectations/NSDictionary%2BSafeExpectations.m#L101-L102) in the original Objective-C code can parse a number to a string, which we don't do here. So, maybe we should use `string(forKey:)` and `number(forKey:)` instead of type casting?